src/github.com/getlantern/flashlight/flashlight.go

```diff
@@ -261,7 +260,14 @@ func runServerProxy(cfg *config.Config) {
 		}
 	}()
 
-	err = srv.ListenAndServe()
+	err = srv.ListenAndServe(func(update func(*server.ServerConfig) error) {
+		err := config.Update(func(cfg *config.Config) error {
+			return update(cfg.Server)
+		})
+		if err != nil {
+			log.Errorf("Error while trying to update: %v", err)
+		}
+	})
```

Comment:
It seems to me like it might be better to just make the srv.register method public and call that after ListenAndServe, as the updateConfig func is only used in register, and we only really want to register after we're 100% sure the server is up (there's a minor, unlikely to be hit race condition with the go routine in ListenAndServe with registering before we're actually serving as far as I can tell).

Does ListenAndServe block though?